ime/app/src/main/java/com/anysoftkeyboard/keyboards/AnyKeyboard.java

```diff
@@ -830,14 +836,16 @@ public AnyKey(
                 }
             }
 
-            // if the shift-character is a symbol, we only show it if the SHIFT is pressed,
-            // not if the shift is active.
-            mShiftCodesAlways =
-                    mShiftedCodes.length == 0
-                            || Character.isLetter(mShiftedCodes[0])
-                            || Character.getType(mShiftedCodes[0]) == Character.NON_SPACING_MARK
-                            || Character.getType(mShiftedCodes[0])
-                                    == Character.COMBINING_SPACING_MARK;
+            if (!mShiftCodesAlwaysOverride) {
```

Comment:
I think you don't need this extra field. You can either convert that into a variable in the method or change the condition to check `mShiftCodesAlways == false`

Reply:
This way `isShiftAlways=false` can also be used to force the non-shift code when shift is only on sticky.

Reply:
right, in that case, you can have `mShiftCodesAlwaysOverride` be a local method variable rather than a class field.